### ISSUE TITLE
Allow setting crossRegionAccessEnabled

### ISSUE THREAD

**Issue by n3v3rf411** [awslabs/aws-java-nio-spi-for-s3#206]
Copying objects from between multiple buckets that are in different regions currently fail with this error:
```
Caused by: software.amazon.awssdk.services.s3.model.S3Exception: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint. (Service: S3, Status Code: 301, Request ID: Y72AK58DSRMM1HN9, Extended Request ID: MISChGBdhF5aNEP+wyhG4FWtUzgnxt4bKvj3uWqLS7pG+t931kUWZWqNVgW3iYFWTiHwWnp18QTFcZXDUdfpLA==)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleErrorResponse(AwsXmlPredicatedResponseHandler.java:156)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleResponse(AwsXmlPredicatedResponseHandler.java:108)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:85)
	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:43)
	at software.amazon.awssdk.core.internal.handler.BaseClientHandler.lambda$successTransformationResponseHandler$7(BaseClientHandler.java:270)
	at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler.lambda$prepare$0(AsyncResponseHandler.java:89)
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966)
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940)
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488)
	at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975)
	at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler$BaosSubscriber.onComplete(AsyncResponseHandler.java:132)
	at software.amazon.awssdk.utils.async.SimplePublisher.doProcessQueue(SimplePublisher.java:275)
	at software.amazon.awssdk.utils.async.SimplePublisher.processEventQueue(SimplePublisher.java:224)
	at software.amazon.awssdk.utils.async.SimplePublisher.complete(SimplePublisher.java:157)
	at java.util.concurrent.CompletableFuture.uniRun(CompletableFuture.java:719)
	at java.util.concurrent.CompletableFuture.uniRunStage(CompletableFuture.java:731)
	at java.util.concurrent.CompletableFuture.thenRun(CompletableFuture.java:2023)
	at software.amazon.awssdk.services.s3.internal.crt.S3CrtResponseHandlerAdapter.onErrorResponseComplete(S3CrtResponseHandlerAdapter.java:143)
	at software.amazon.awssdk.services.s3.internal.crt.S3CrtResponseHandlerAdapter.handleError(S3CrtResponseHandlerAdapter.java:132)
	at software.amazon.awssdk.services.s3.internal.crt.S3CrtResponseHandlerAdapter.onFinished(S3CrtResponseHandlerAdapter.java:101)
	at software.amazon.awssdk.crt.s3.S3MetaRequestResponseHandlerNativeAdapter.onFinished(S3MetaRequestResponseHandlerNativeAdapter.java:24)
```

Enabling crossRegionAccessEnabled fixes this. It would be great to set this i.e. through env when creating the filesystem.

**Comment by n3v3rf411** [comment]
One way to get around this is to set the default client builder:
```scala
fs.clientProvider()
  .asyncClientBuilder(
    S3AsyncClient.crtBuilder.crossRegionAccessEnabled(true)
  )
```

**Comment by markjschreiber** [comment]
I would prefer not to do this by default because of this warning in the SDK. 

> Enabling this mode has several drawbacks, as it can increase latency if the bucket's location is physically far from the location of the request.Therefore, it is strongly advised, whenever possible, to know the location of your buckets and create a region-specific client to access them

Cross region data transfer can also incur additional costs which is something that a user should probably opt in for.

However, we should certainly allow this and being able to set this through the env seems like a great way to do this.

Would you be open to adding a PR for this?

**Comment by n3v3rf411** [comment]
I just remembered about these issues that I filed. I'll take a stab at this one.

**Comment by markjschreiber** [comment]
Thinking on this some more, perhaps cross region access should be enabled by default. It is for the `aws s3` commands from the CLI and it can be difficult to know that a bucket is in a different region so for maximum ease of use I think we should do this.

**Comment by markjschreiber** [comment]
Fixed by #318
